### ISSUE TITLE
https://www.nsupdate.info/  The account registration is closed.

### ISSUE THREAD

**Issue by stokito** [nsupdate-info/nsupdate.info#496]
I can't sign up into the site. How long the registration will be closed?

**Comment by ThomasWaldmann** [comment]
I won't reopen it (== self service account creation) in the near future. There is just too much abuse by criminals and neither want to support their actions nor deal with deleting their stuff all the time.

I might think about a procedure for personally requesting an account though.

**Comment by stokito** [comment]
Interesting how other DDNS providers working with abuses. In the DuckDNS FAQ said 

> Q: Do you have an Abuse policy?
> 
> A: yes, we will block, as much as we can anyone who is abusing our service, see our [Terms of Use](https://www.duckdns.org/tac.jsp) and [Privacy Policy](https://www.duckdns.org/pp.jsp). If you have any issues please use the [Google Group](https://groups.google.com/forum/#!forum/duckdns) to contact us

https://www.duckdns.org/faqs.jsp

Maybe this is solvable somehow.

The bigger problem is for tunnels providers.
I remember this article https://pagekite.net/2018-07-18/Phishing_and_spam
The Ngrok also spent a lot efforts to stop spam
https://ngrok.com/abuse
here is some examples
https://medium.com/@g33xter/phishing-with-ngrok-252309890b87

Maybe their experience may be useful

**Comment by stefanbaur** [comment]
Hi. I already have (or is that now "had?") an account, which was handled via Twitter auth. However, when I try to log in, I get "Server Error (error 500)". Is this intentional or an unwanted side effect of you closing new account registration?

**Comment by ThomasWaldmann** [comment]
external auth is disabled due to abuse by criminals.

please write me an email, give your login name and email you used and a few hostnames you have and i will set a password for you.

**Comment by stefanbaur** [comment]
Sent (to the address listed in your github profile), please check your inbox. :)

**Comment by jgsieve** [comment]
> external auth is disabled due to abuse by criminals. please write me an email, give your login name and email you used and a few hostnames you have and i will set a password for you.

emailed you, thanks!

**Comment by sporkman** [comment]
> Interesting how other DDNS providers working with abuses. In the DuckDNS FAQ said
> 
> > Q: Do you have an Abuse policy?
> > A: yes, we will block, as much as we can anyone who is abusing our service, see our [Terms of Use](https://www.duckdns.org/tac.jsp) and [Privacy Policy](https://www.duckdns.org/pp.jsp). If you have any issues please use the [Google Group](https://groups.google.com/forum/#!forum/duckdns) to contact us
> 
> https://www.duckdns.org/faqs.jsp
> 
> Maybe this is solvable somehow.
> 
> The bigger problem is for tunnels providers. I remember this article https://pagekite.net/2018-07-18/Phishing_and_spam The Ngrok also spent a lot efforts to stop spam https://ngrok.com/abuse here is some examples https://medium.com/@g33xter/phishing-with-ngrok-252309890b87
> 
> Maybe their experience may be useful

Running a free service like this AND keeping it clear of people trying to use for C&C servers and other nefarious purposes, and then also fighting DDoS attacks from both good guys and bad guys is a full-time job. In theory, sure, any of us could set this up on a bunch of VPSs out there and advertise it as "free dynamic dns service", but without a small team of people constantly babysitting it, it's just not practical.

Just glad the software exists so I can self-host (and I'm looking at this because DuckDNS is again intermittent all weekend long).

**Comment by luckman212** [comment]
@sporkman Is there a fairly simple guide or any steps on setting up a selfhosted instance from scratch on a VPS (DigitalOcean etc)?

**Comment by MartinX3** [comment]
I used the email mentioned on the website for registration.
I wonder if I should also use the email of his github profile or what his average response times are?

**Comment by mpathy** [comment]
Hi Thomas! I have used the service before, I used my github account to login. This isnt possible anymore. Password reset sends me no password. I think it is because I never had a real password, but instead GitHub oAuth. But GitHub oAuth on your site is broken.

TL;DR - I have an account on nsupdate already, but cant login because of broken Github login, what shall i do?

**Comment by ThomasWaldmann** [comment]
Hey mpathy, long time no talking!

Just send me an email from the email address used in the account, also give github username and a few of your dyndns hostnames).

I'll then set a temporary pw for your account, so you can log in locally.

**Comment by leewp14** [comment]
> Hey mpathy, long time no talking!

Hi @ThomasWaldmann, sorry to jump in, I was facing difficulties to login my account using Google OAuth as it says the following error: 
![image](https://user-images.githubusercontent.com/17796518/206390012-95f931f5-b6ff-4f4f-ae65-d071dbbd62c2.png)
Can I also get your assistance on this or should I contact the support email? Because I have a number of dormant hosts and nsupdate has been asking me to delete them since it is no longer active. 
Thanks in advance!

**Comment by ThomasWaldmann** [comment]
@leewp14 https://github.com/nsupdate-info/nsupdate.info/issues/496#issuecomment-1187319496 see there.

Yes, as existing account owner just write me an email (from the email address you use for nsupdate.info), give your username and some existing host names of yours and I'll send you a temp password for direct login.

**Comment by Pysis868** [comment]
Add indication on the site that the external login methods are no longer supported, such as removing the link and fading the buttons?

**Comment by stokito** [comment]
@ThomasWaldmann could you please explain how criminals are using the service?
As far I understood they are making a phishing like bank.nsupdate.info or they just need for any domain for their website?
In first case we may make a filter by registred trademarks and banned words. And we may require for domain to start from bad so that users will see badbank.nsupdate.info and this should make them more cautios.

But if criminals just need for any domain and they don't want to register because it needs for a credit card then this makes it more dificult. The only solution would be to require a payment of 1$ and then rollback.
Maybe you can create a group of moderators voulontiers who can review emails about abuses. How many of abuses letters do you receive?

I'm asking because I made my own ddns server and I need to know how to protect.

I decided to use a different approach. I want to make my users to be automatically registered. So they'll just generate a long random domain (uuid or ed25519 pub key) and token and configure a ddns client. The new domain with the token are registred and any next updates will require of the same password.
Since the domains are not human readable then nobody interested for a sqoting. Not used domains will be removed after a period.

**Comment by ThomasWaldmann** [comment]
Well, as far as i could see / got notified of:
- they register an arbitrary dyndns hostname to point to a distribution host of illegal material (e.g. CSAM), malware (computer virus/trojan infected stuff) or copyright violations (ebooks, movies, ...)
- they register specific dyndns hostnames to point to fishing sites, like theirbank.nsupdate.info, theirpostalservice.nsupdate.info - they are rather creative with these so it is hard to blacklist them, even when using regexes.
- they register specific dyndns hostnames to sell fake / questionable products
- they register less specific dyndns hostnames for other sites with services for their criminal operations.

I think a simple list of trademarks / registered names would be huge and won't solve the problem. E.g. if you had wellsfargo (a bank) on your list, the abuser would just use wells-fargo or we11sfargo or wellsfarg0 or ... - some are even hard to match with regexes.

I thought about obligatory payments. That might deter some of the criminals, because that would harm their anonymity and some kind of information about them might be in reach for criminal investigators then. But I guess even that could be worked around, just by abusing the credit card of someone else. Also, I don't like that, because it would basically turn the free service into a commercial one, cause more work for me and would be either for nothing or would be more expensive...

There is no "moderation admin interface" (yet?) for the nsupdate.info software (I currently do that via the django admin). In any case, I would only give access to very trusted persons I personally know.

Using random (like uuid) hostnames and not allowing the user to choose the hostname removes some of the criminal use cases, but not all. Of course, it also removes some of the reasons why legitimate users want to have a dyndns name in the first place (== because they can remember it easily). It also removes some pattern matching options for the admin to fight criminal abusers.

About removing inactive hostnames after some time: we also do that (after a rather long time), but it first notifies the users multiple times via email about that this is going to happen if there is no activity. This is not completely unproblematic, because sometimes users give fake email addresses or just don't get or read our emails. Hostnames that are not updated for longer times usually happen for cable internet providers (IP does not change).

It is a good practice to not update the dyndns host if the IP did not change, BUT still send 1 monthly **unconditional** update just to signal "hey, I am still alive and using this dyndns host".

About self-registration, oauth (external accounts):
- when only using local accounts on nsupdate.info and disallowing self-registration, i have some control over who can create dyndns hosts and who not. if some user gets on the radar by creating scam hostnames, i can delete the user account and all their hosts and they won't get in again easily.
- when allowing self-registration, a user who was removed can easily just create a new account (using a new name / email address) and can continue abusing the service until they get noticed again. this is an endless whack-a-mole game...
- when allowing external accounts (oauth, twitter, github, ... logins) it even gets more comfortable for the abusers, because they can maintain e.g. a twitter account (which i have no control over) and the nsupdate.info service then would just auto-create a profile for them based on that twitter account. even if i removed their hosts and their profile, they could immediate get in again and just recreate their hosts.

Since I have removed self-registration, the amount of abuse has significantly decreased (also the amount of abuse notifications I get via email).

But I usually can't keep up with new user registration request emails (also hard to know how they will use the service), so in the past months I focussed on helping existing users to get into their accounts again (e.g. if they only had used the external auth [disabled now], but did not set a local password).

**Comment by ThomasWaldmann** [comment]
Update: I updated the 2nd post with the current state of affairs.

Also cleaned up this ticket a bit, remove posts that are not useful any more.

**Comment by dd8zc** [comment]
Very sad. Unfortunately, my friends recommended the service to me too late. If registration becomes possible again (I can also verify myself by personally sending stamps, even though no one uses stamps anymore), it would be cool if this thread could be updated.

**Comment by Mark3xtrm** [comment]
So nsupdate is dead or not? As far as I can see, new registrations are currently and still not possible (?)

![image](https://github.com/user-attachments/assets/e5371a2c-d0a8-44f3-b491-89bb19bfcea8)

**Comment by abcummins33** [comment]
Hi @ThomasWaldmann, if possible could you create an account for me? Let me know if there is any additional information you would need from me.
